src/google/protobuf/compiler/cpp/cpp_file.cc

```diff
@@ -599,7 +599,7 @@ void FileGenerator::GenerateBuildDescriptors(io::Printer* printer) {
     // string literal.
     if (breakdown_large_file && file_data.size() > 65535) {
       printer->Print(
-          "static const char descriptor[] = {\n");
+          "static const signed char descriptor[] = {\n");
```

Comment:
Do you mean when char is unsigned, 'const char value[] = "xxx"' will stop working?